numpy/_core/src/multiarray/methods.c

```diff
@@ -919,7 +919,11 @@ array_wraparray(PyArrayObject *self, PyObject *args)
                 PyArray_NDIM(arr),
                 PyArray_DIMS(arr),
                 PyArray_STRIDES(arr), PyArray_DATA(arr),
-                PyArray_FLAGS(arr), (PyObject *)self, obj);
+                PyArray_FLAGS(arr), (PyObject *)self, (PyObject *)arr);
+    }
+    else if (return_scalar) {
```

Comment:
Absolute nitpick, but for similarity with the scalar case, perhaps
```
Py_INCREF(arr)
if (return_scalar) {
    ...
```
Since the previous clause had a return, we don't need an `else if` strictly.

Though arguably less clear, so feel free to ignore (really, just because I thought I saw  a difference in ref counting when reviewing).